src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -1,43 +1,217 @@
+#nullable enable
+
 using System.Net;
 using System.Net.Http.Headers;
-using GraphQL.Instrumentation;
+using GraphQL.Server.Transports.AspNetCore.Errors;
 using GraphQL.Transport;
 using GraphQL.Types;
+using GraphQL.Validation;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.DependencyInjection;
 
 namespace GraphQL.Server.Transports.AspNetCore;
 
-/// <summary>
-/// ASP.NET Core middleware for processing GraphQL requests. Can processes both single and batch requests.
-/// See <see href="https://www.apollographql.com/blog/query-batching-in-apollo-63acfd859862/">Transport-level batching</see>
-/// for more information. This middleware useful with and without ASP.NET Core routing.
-/// <br/><br/>
-/// GraphQL over HTTP <see href="https://github.com/APIs-guru/graphql-over-http">spec</see> says:
-/// GET requests can be used for executing ONLY queries. If the values of query and operationName indicates that
-/// a non-query operation is to be executed, the server should immediately respond with an error status code, and
-/// halt execution.
-/// <br/><br/>
-/// Attention! The current implementation does not impose such a restriction and allows mutations in GET requests.
-/// </summary>
-/// <typeparam name="TSchema">Type of GraphQL schema that is used to validate and process requests.</typeparam>
-public class GraphQLHttpMiddleware<TSchema> : IMiddleware
+/// <inheritdoc/>
+/// <typeparam name="TSchema">
+/// Type of GraphQL schema that is used to validate and process requests.
+/// Specifying <see cref="ISchema"/> will pull the registered default schema.
```

Comment:
What is _default schema_ ? In addition, one gets the impression that otherwise (in case of typed schema) the schema will not be pulled.

Reply:
The first schema registered via `AddSchema`.  See the `AddSchema` documentation here:

https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/GraphQLBuilderExtensions.cs#L152-L161

> In addition, one gets the impression that otherwise (in case of typed schema) the schema will not be pulled.

It seems rather clear to me -- "Type of GraphQL schema that is used to validate and process requests.".  If you would like to clarify the comment somehow, feel free to write a suggestion.